### PR TITLE
Add UTM data to semantic URLs. (CHAN-71)

### PR DESCRIPTION
**Pull request by wting**
Confirmed locally:
```shell
╭─@eikyo ~ ‹python-2.7.12›
╰─ curl -s -A "wting-${RANDOM}" http://localhost:4444/r/explainlikeimfive/comments/4ukacd/eli5_why_college_is_so_high_priced_in_the_us/ | python -c 'import sys; from BeautifulSoup import BeautifulSoup; print(BeautifulSoup(sys.stdin.read()).findAll("script", type="application/ld+json")[0].contents[0])' | jq '{url, sharedContent}'
{
  "url": "http://localhost:4444/r/explainlikeimfive/comments/4ukacd/eli5_why_college_is_so_high_priced_in_the_us/?utm_source=search&utm_medium=structured_data",
  "sharedContent": {
    "url": "http://localhost:4444/r/explainlikeimfive/comments/4ukacd/eli5_why_college_is_so_high_priced_in_the_us/?utm_source=search&utm_medium=structured_data"
  }
}
```

👓  @prashtx 

cc: @drunken-economist @schwers

### CODE REVIEW

File: test/lib/extractErrorMsg.es6.js [hunk 1]
```diff
@@ -9,26 +9,26 @@ const GENERIC_MESSAGE = 'There was a problem';
...
 
 describe('lib: extractErrorMsg', () => {
-  it('returns error messages nested in a compolete, 2-deep array', () => {
+  it('returns error messages nested in a complete, 2-deep array', () => {
     expect(extractErrorMsg({errors: [['error type', TEST_MESSAGE]]}))
       .to.equal(TEST_MESSAGE);
   });
-  
+
   it('returns a any message found in error.message', () => {
```
Comment (drunken-economist) [inline test/lib/extractErrorMsg.es6.js]: typo, not a huge deal though 'returns a any message'

**Comment by drunken-economist** [comment]
🐟

**Comment by prashtx** [comment]
In cases where `sharedContent.url` points to a Reddit page, I think we should also add the UTM parameters. I think you can piggyback on the existing conditions in the handling of `sharedContent`.

💅

**Comment by wting** [comment]
- cc'ed @schwers since @ajacksified is on PTO.
- added support for `sharedContent.url`
- added relevant tests

💇

**Comment by wting** [comment]
- fix linter complaints

💇

File: src/views/components/GoogleCarouselMetadata.jsx [hunk 2]
```diff
@@ -4,6 +4,27 @@ import has from 'lodash/object/has';
...
+export function isRedditDomain (hostname) {
+  return hostname.endsWith('reddit.com');
+}
+
+/*
+ * Adds UTM parameters to track inbound links from search engines.
+ */
+export function addUtmTracking (url_string) {
+  const urlObject = url.parse(url_string, true);
+  if (!url_string || !isRedditDomain(urlObject.hostname)) {
```
Comment (prashtx) [inline src/views/components/GoogleCarouselMetadata.jsx]: Is it possible for `url_string` to be `null`/`undefined`, or are you guarding against empty string? `url.parse` will throw an exception if it receives null/undefined, so if those are possibilities, we should test before that call.

An empty string will lead to a null value for `urlObject.hostname`, so if that's plausible, then it's fair to test here.

**Comment by prashtx** [comment]
1 small question about null/undefined/empty handling.

💅

File: src/views/components/GoogleCarouselMetadata.jsx (hunk 2 above)
Comment (wting) [inline src/views/components/GoogleCarouselMetadata.jsx]: Good point, the original intent was to guard against an empty string.

- `baseObject.url` will always be populated.
- `baseObject.sharedContent.url` will exist or be an empty string due to [`mobilify`](https://github.com/reddit/reddit-mobile/blob/8767a9747f4a5e03091b0f9729db1a298131a214/src/lib/mobilify.es6.js#L3).

It doesn't hurt to write defensive code, so I'll add null checking anyway.

**Comment by wting** [comment]
- add `null` handling and related test

💇

**Comment by prashtx** [comment]
slick.

🐟

**Comment by wting** [comment]
Merged into 32fbd83d.
